client/ayon_usd/hooks/usd_pinning_root.py

```diff
@@ -1,23 +1,26 @@
 """Pre-launch hook to set USD pinning related environment variable."""
-from typing import ClassVar
-
+from ayon_api import get_project
 from ayon_applications import LaunchTypes, PreLaunchHook
+from ayon_core.pipeline import Anatomy
 
 
 class UsdPinningRoot(PreLaunchHook):
     """Pre-launch hook to set USD_ROOT environment variable."""
 
-    app_groups: ClassVar = {"maya", "houdini", "blender", "unreal"}
+    app_groups = {"maya", "houdini", "blender", "unreal"}
     # this should be set to farm_render, but this issue
     # https://github.com/ynput/ayon-applications/issues/2
     # stands in the way
-    launch_types: ClassVar = {LaunchTypes.farm_publish}
+    launch_types = {LaunchTypes.farm_publish}
 
     def execute(self) -> None:
         """Set environments necessary for pinning."""
-        if self.launch_context.env.get("PINNING_FILE_PATH"):
+        if not self.launch_context.env.get("PINNING_FILE_PATH"):
             return
-        anatomy = self.launch_context.anatomy
+
+        project_entity = get_project(self.data.project_name)
+        anatomy = Anatomy(
+            self.data.project_name, project_entity=project_entity)
```

Comment:
`self.data["anatomy"]` should just work fine?

```suggestion
        anatomy = self.data["anatomy"]
```

The Pre OCIO hook uses that - [also on the farm](https://github.com/ynput/ayon-core/blob/f827dc2060488de74421cd7d0b8fc826a499975e/client/ayon_core/hooks/pre_ocio_hook.py#L44)?

Reply:
You may need to specify `order = 0` but I don't think you need to.... because without order I believe it runs last?

Reply:
you're right, before it was trying to use anatomy from launch context but it isn't there. Would be great to somehow unify this.